From 2a3fddf592000765242a8fedbe348e07f7923393 Mon Sep 17 00:00:00 2001
From: Jordan Lazare <Jordan.Lazare@amd.com>
Date: Tue, 4 Oct 2016 12:27:12 -0400
Subject: [PATCH 1175/1722] drm/amd/dal: Fix bios powergating call to use
 correct structure

Using the non-allocation version of the structure leads to stack corruption
when booting with eDP panels among other possible bad situations

Change-Id: I2a2f4cb77ade7db8040800baf379790f1570d366
Signed-off-by: Jordan Lazare <Jordan.Lazare@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/bios/command_table.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/bios/command_table.c b/drivers/gpu/drm/amd/dal/dc/bios/command_table.c
index 6836866..51f6052 100644
--- a/drivers/gpu/drm/amd/dal/dc/bios/command_table.c
+++ b/drivers/gpu/drm/amd/dal/dc/bios/command_table.c
@@ -2515,7 +2515,7 @@ static enum bp_result enable_disp_power_gating_v2_1(
 {
 	enum bp_result result = BP_RESULT_FAILURE;
 
-	ENABLE_DISP_POWER_GATING_PARAMETERS_V2_1 params = {0};
+	ENABLE_DISP_POWER_GATING_PS_ALLOCATION params = {0};
 	uint8_t atom_crtc_id;
 
 	if (bp->cmd_helper->controller_id_to_atom(crtc_id, &atom_crtc_id))
-- 
2.7.4

